mammoth/translate/translator.py

```diff
@@ -787,22 +789,29 @@ def translate_batch(self, batch, src_vocabs, attn_debug):
                     stepwise_penalty=self.stepwise_penalty,
                     ratio=self.ratio,
                     ban_unk_token=self.ban_unk_token,
+                    device=self._device,
                 )
             return self._translate_batch_with_strategy(batch, src_vocabs, decode_strategy)
 
-    def _run_encoder(self, batch):
-        src, src_lengths = batch.src if isinstance(batch.src, tuple) else (batch.src, None)
-
-        enc_states, memory_bank, src_lengths, mask = self.model.encoder(
-            src, src_lengths
+    def _run_encoder(self, active_encoder, batch):
+        src = rearrange(batch.src.tensor, 't b 1 -> b t')
```

Comment:
maybe we should work towards maintaining a convention throughout the code (e.g, batch x seq x feats)

Reply:
Yes, it would be great to standardize on the x-transformers preferred order, to avoid unnecessary transposes. Might even have a speed impact.

This should be a pure refactoring, so we can test that the post-refactoring produces exactly the same results.

Reply:
separate ticket, then?